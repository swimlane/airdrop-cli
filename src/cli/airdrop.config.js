module.exports = {
  defaults: {
    package_path: './-/',
    package_root: '/-/',    
  },
  name: 'airdrop'
}