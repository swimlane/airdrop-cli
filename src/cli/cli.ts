import { build, GluegunToolbox } from 'gluegun';

/**
 * Create the cli and kick it off
 */
export async function run(argv?: string[] | string): Promise<GluegunToolbox> {
  // create a CLI runtime
  const cli = build('airdrop')
    .src(__dirname)
    .help()
    .version()
    .exclude([
      'semver',
      'system', 
      'prompt',
      'http',
      'template',
      'patching'
    ])
    .defaultCommand({
      run: async ({ print: { info }, meta: { version }, runtime: { brand } }: GluegunToolbox) => {
        info(`${brand} version ${version()}`)
        info(``)
        info(`  Type ${brand} --help for more info`)
      }
    })
    .command({
      name: 'add-bundle',
      alias: ['ab'],
      run: async (toolbox: GluegunToolbox) => {
        const { parameters, print } = toolbox;
        const force = parameters.options.force ? '--force' : '';
        const optimize = parameters.options.optimize ? '--optimize' : '';

        print.info('Adding packages');
        await cli.run(`a ${parameters.string} ${force}`);

        print.info('Bundling packages');
        await cli.run(`b ${parameters.string} ${force} ${optimize}`);
      }
    })
    .create();

  return cli.run(argv);
}
