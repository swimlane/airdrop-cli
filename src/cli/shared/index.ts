export * from './getMap';
export * from './addPackages';
export * from './bundlePackages';