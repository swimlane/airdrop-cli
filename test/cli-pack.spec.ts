import { createSandbox, execPkgdrop } from './cli.util';

const TIMEOUT = 100000;

describe('pack', () => {
  let sandbox: any;
  let output: string;

  beforeAll(async () => {
    sandbox = await createSandbox();
    await execPkgdrop(`init -y --offline`);
    await execPkgdrop(`add lit-element@2.0.0 --clean`);
    output = await execPkgdrop(`pack pkgdrop-pack-test.tgz`);
  }, TIMEOUT);

  afterAll(async () => {
    sandbox.clean();
  }, TIMEOUT);

  test('displays console messages', () => {
    expect(output).toMatchSnapshot();
  });

  test('files exist', async () => {
    expect(await sandbox.exists('pkgdrop-pack-test.tgz')).toBe('file');
  });

  test('importmap', async () => {
    expect(await sandbox.read('-/importmap.json')).toMatchSnapshot();
  });

  describe('merge', () => {
    beforeAll(async () => {
      await execPkgdrop(`add lit-element@2.1.0 --clean`);
      output = await execPkgdrop(`merge pkgdrop-pack-test.tgz`);
    }, TIMEOUT);

    test('displays console messages', () => {
      expect(output).toMatchSnapshot();
    });

    test('files', async () => {
      expect(await sandbox.exists('-/lit-element@2.1.0')).toBe('dir');
      expect(await sandbox.exists('-/lit-element@2.0.0')).toBe('dir');
    });

    test('importmap', async () => {
      expect(await sandbox.read('-/importmap.json')).toMatchSnapshot();
    });
  });
});