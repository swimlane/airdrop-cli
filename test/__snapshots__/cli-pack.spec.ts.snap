// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pack displays console messages 1`] = `
"Writing pkgdrop-pack-test.tgz
Done! [XX s]"
`;

exports[`pack dryrun 1`] = `
"Writing pkgdrop-pack-test.tgz [dry run]
Done! [XX s]"
`;

exports[`pack importmap 1`] = `
"{
  \\"imports\\": {
    \\"lit-element@2.0.0\\": \\"/-/lit-element@2.0.0/lit-element.js\\",
    \\"lit-element@2.0.0/\\": \\"/-/lit-element@2.0.0/\\",
    \\"lit-element@2\\": \\"/-/lit-element@2.0.0/lit-element.js\\",
    \\"lit-element@2/\\": \\"/-/lit-element@2.0.0/\\"
  },
  \\"scopes\\": {
    \\"lit-element@2.0.0/\\": {
      \\"lit-html\\": \\"/-/lit-html@1.1.2/lit-html.js\\",
      \\"lit-html/\\": \\"/-/lit-html@1.1.2/\\"
    },
    \\"lit-html@1.1.2/\\": {}
  }
}"
`;

exports[`pack merge displays console messages 1`] = `
"Extracting pkgdrop-pack-test.tgz
Writing importmap
Done! [XX s]"
`;

exports[`pack merge importmap 1`] = `
"{
  \\"imports\\": {
    \\"lit-element@2.1.0\\": \\"/-/lit-element@2.1.0/lit-element.js\\",
    \\"lit-element@2.1.0/\\": \\"/-/lit-element@2.1.0/\\",
    \\"lit-element@2\\": \\"/-/lit-element@2.1.0/lit-element.js\\",
    \\"lit-element@2/\\": \\"/-/lit-element@2.1.0/\\",
    \\"lit-element@2.0.0\\": \\"/-/lit-element@2.0.0/lit-element.js\\",
    \\"lit-element@2.0.0/\\": \\"/-/lit-element@2.0.0/\\"
  },
  \\"scopes\\": {
    \\"lit-element@2.1.0/\\": {
      \\"lit-html\\": \\"/-/lit-html@1.1.2/lit-html.js\\",
      \\"lit-html/\\": \\"/-/lit-html@1.1.2/\\"
    },
    \\"lit-html@1.1.2/\\": {},
    \\"lit-element@2.0.0/\\": {
      \\"lit-html\\": \\"/-/lit-html@1.1.2/lit-html.js\\",
      \\"lit-html/\\": \\"/-/lit-html@1.1.2/\\"
    }
  }
}"
`;
